### PR TITLE
Updating style guide for es2015 rules

### PR DESCRIPTION
**Pull request by jcreamer898**
Adding in rules for working with ES2015. This is totally up for any collaboration.

### CODE REVIEW

**Comment by davidthesavage** [comment]
:+1:

File: app/docs/js/code-style.md [hunk 1]
```diff
@@ -147,4 +124,155 @@ $(selector)
...
+], function(Godliness, Cleanliness) {
+```
+
+## ES6 (2015) Guide
+
+### Variable Declarations
+
+* Use `let` almost always, and declare in correct scope. No more need to hoist all declarations
+
+* Use `const` when needing a **primitive** value (aka string, number, boolean) to never change
```
Comment (stevebrownlee) [inline app/docs/js/code-style.md]: Why only primitives? I can definitely see the case where you want a constant object declaration that can be modified, but not be redefined with another object.

Comment (jcreamer898) [inline app/docs/js/code-style.md]: Will modify, and mention some things about it. Thanks.

File: app/docs/js/code-style.md [hunk 2]
```diff
@@ -147,4 +124,155 @@ $(selector)
...
+let name = this.user.name,
+    age = this.user.age,
+    location = this.user.location;
+
+// good
+let { name, age, location } = this.user;
+```
+
+### Arrow functions
+* Use arrow functions possible
```
Comment (stevebrownlee) [inline app/docs/js/code-style.md]: Option 1: "Use arrow functions **whenever** possible"

Option 2: "Arrow functions should be used by default, except in rare cases where it's not possible"

Comment (jcreamer898) [inline app/docs/js/code-style.md]: :+1:

**Comment by stevebrownlee** [comment]
I would also like you to add a section about using Array Comprehensions whenever simple transformations or filtering needs to take place on an array.

**Comment by jcreamer898** [comment]
@stevebrownlee you mean like `[for (i of numbers) i*i ]` or `map`, `filter`, etc?

**Comment by stevebrownlee** [comment]
for...of

**Comment by stevebrownlee** [comment]
Squerge :+1:

**Pull request merged**